proposals/0257-string-uninitialized-initializer.md

```diff
@@ -0,0 +1,141 @@
+  /// replacement character (`"\u{FFFD}"`); This may require resizing
+  /// the buffer beyond its original capacity.
+  ///
+  /// The following examples use this initializer with the contents of two
+  /// different `UInt8` arrays---the first with well-formed UTF-8 code unit
+  /// sequences and the second with an ill-formed sequence at the end.
+  ///
+  ///     let validUTF8: [UInt8] = [67, 97, 102, -61, -87, 0]
+  ///     let s = String(unsafeUninitializedCapacity: validUTF8.count,
+  ///                    initializingUTF8With: { (ptr, count) in
+  ///         ptr.initializeFrom(validUTF8)
+  ///         count = validUTF8.count
+  ///     })
+  ///     // Prints "Optional(Café)"
+  ///
+  ///     let invalidUTF8: [UInt8] = [67, 97, 102, -61, 0]
+  ///     let s = String(unsafeUninitializedCapacity: invalidUTF8.count,
+  ///                    initializingUTF8With: { (ptr, count) in
+  ///         ptr.initializeFrom(invalidUTF8)
+  ///         count = invalidUTF8.count
+  ///     })
+  ///     // Prints "Optional(Caf�)"
+  ///
+  ///     let s = String(unsafeUninitializedCapacity: invalidUTF8.count,
+  ///                    initializingUTF8With: { (ptr, count) in
+  ///         ptr.initializeFrom(invalidUTF8)
+  ///         count = 0
+  ///     })
+  ///     // Prints "Optional("")"
+  ///
+  /// - Parameters:
+  ///   - capacity: The number of UTF-8 code units worth of memory to allocate
+  ///       for the String.
+  ///   - initializer: A closure that initializes elements and sets the count of
+  ///       the new String
+  ///     - Parameters:
+  ///       - buffer: A buffer covering uninitialized memory with room for the
+  ///           specified number of UTF-8 code units.
+  ///       - initializedCount: Set this to the number of elements in `buffer`
+  ///           that were actually initialized by the `initializer`
+  @inlinable @inline(__always)
+  public init(
+    unsafeUninitializedCapacity capacity: Int,
+    initializingUTF8With initializer: (
+    _ buffer: UnsafeMutableBufferPointer<UInt8>,
+    _ initializedCount: inout Int
+    ) throws -> Void
+  ) rethrows
+```
+
+### Specifying a capacity
+
+The initializer takes the specific capacity that a user wants to work with as a
+parameter. The buffer passed to the closure has a count that is exactly the
+same as the specified capacity, even if the ultimate size of the new `String` is larger.
+
+### Guarantees after throwing
+
+Because `UTF8.CodeUnit` is a trivial type, there are no special considerations about the state of the buffer when an error is thrown, unlike `Array`.
+
+## Source compatibility
+
+This is an additive change to the standard library,
+so there is no effect on source compatibility.
+
+## Effect on ABI stability
+
+The new initializer will be part of the ABI, and will result in calls to a new @usableFromInline symbol being inlined into client code. Use of the new initializer is gated by @availability though, so there's no back-deployment concern.
```

Comment:
Above had `@inlinable`, and "usableFromInline symbol being inlined into client code" sounds like `@inlinable`. Is that what you meant?

Reply:
There's a new usableFromInline symbol as well